quarkus.ls.ext/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/extensions/ollama/OllamaItemMetadataProvider.java

```diff
@@ -0,0 +1,134 @@
+
+import static com.redhat.quarkus.extensions.ollama.OllamaConstants.DEFAULT_OLLAMA_BASE_URL;
+import static com.redhat.quarkus.extensions.ollama.OllamaConstants.QUARKUS_LANGCHAIN4J_OLLAMA_BASE_URL_KEY;
+import static com.redhat.quarkus.extensions.ollama.OllamaConstants.QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.lsp4mp.commons.metadata.ItemHint;
+import org.eclipse.lsp4mp.commons.metadata.ItemMetadata;
+import org.eclipse.lsp4mp.commons.metadata.ValueHint;
+import org.eclipse.lsp4mp.commons.utils.StringUtils;
+import org.eclipse.lsp4mp.extensions.ExtendedMicroProfileProjectInfo;
+import org.eclipse.lsp4mp.extensions.ItemMetadataProvider;
+import org.eclipse.lsp4mp.model.Node;
+import org.eclipse.lsp4mp.model.Node.NodeType;
+import org.eclipse.lsp4mp.model.PropertiesModel;
+import org.eclipse.lsp4mp.model.Property;
+
+import com.redhat.quarkus.extensions.ollama.client.OllamaClient;
+
+/**
+ * Properties provider to collect available Ollama model values for the property
+ * 'quarkus.langchain4j.ollama.chat-model.model-id'.
+ * 
+ * The Ollama models are collected by using Ollama Search Api by using the
+ * following base url:
+ * 
+ * <ul>
+ * <li>defined with 'quarkus.langchain4j.ollama.base-url' property.
+ * <li>otherwise with 'http://localhost:11434'</li>
+ * </ul>
+ */
+public class OllamaItemMetadataProvider implements ItemMetadataProvider {
+
+	private ExtendedMicroProfileProjectInfo projectInfo;
+	private boolean available;
+
+	public OllamaItemMetadataProvider(ExtendedMicroProfileProjectInfo projectInfo) {
+		this.projectInfo = projectInfo;
+		// the Ollama models are collected only if
+		// 'quarkus.langchain4j.ollama.chat-model.model-id' property exists.
+		this.available = this.projectInfo.getProperties().stream()
+				.anyMatch(p -> QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY.equals(p.getName()));
+		if (available) {
+			ItemHint hint = new ItemHint();
+			hint.setName(QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY);
+			hint.setValues(Collections.emptyList());
+			projectInfo.getHints().add(hint);
+		}
+	}
+
+	@Override
+	public boolean isAvailable() {
+		return available;
+	}
+
+	@Override
+	public void update(PropertiesModel document) {
+		// Called when application.properties file is opened or updated (when user type
+		// something in the file).
+		boolean hasModelProperty = getProperty(document, QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY) != null;
+		ItemHint hint = projectInfo.getHint(QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY);
+		if (hasModelProperty && hint != null) {
+			// The application.properties declare the
+			// 'quarkus.langchain4j.ollama.chat-model.model-id' property,
+			// we need to collect available Ollama models by using Ollama Search Api
+			String baseUrl = getSearchBaseUrl(document);
+			OllamaClient client = OllamaClient.getInstance();

```

Comment:
I would not use a singleton here. If you have 2 quarkus projects pointing to 2 different ollama instances, this will absolutely be an issue

Reply:
fixed